components/theia/packages/gitpod-extension/src/node/supervisor-client-provider.ts

```diff
@@ -0,0 +1,17 @@
+import { injectable } from "inversify";
+import { IStatusServiceClient, StatusServiceClient } from "@gitpod/supervisor/lib/status_grpc_pb";
+import * as grpc from "@grpc/grpc-js";
+
+@injectable()
+export class SupervisorClientProvider {
+    protected statusClient: IStatusServiceClient | undefined;
+    
+    public async getStatusClient(): Promise<IStatusServiceClient> {
+        if (!this.statusClient) {
+            this.statusClient = new StatusServiceClient("localhost:22999", grpc.credentials.createInsecure());
```

Comment:
is it fine to hardcode or there is process.env variable for port?

Reply:
That's a good point. Considering that Theia/the IDE is started by supervisor, it would make sense to set an environment variable.

Reply:
I've introduced a `SUPERVISOR_ADDR` env var which supervisor passes to the IDE